packages/utils/test/unit/promiserace.test.ts

```diff
@@ -14,91 +15,67 @@ describe("racePromisesWithCutoff", () => {
       setTimeout(() => reject(Error(value)), delay);
     });
 
-  // For brevity in testcases
-  const precutoff = RaceEvent.precutoff;
-  const cutoff = RaceEvent.cutoff;
-  const resolvedatcutoff = RaceEvent.resolvedatcutoff;
-  const pretimeout = RaceEvent.pretimeout;
-  const timeout = RaceEvent.timeout;
-  const resolved = RaceEvent.resolved;
-  const rejected = RaceEvent.rejected;
-
-  // Second item in testcase row i.e. array of numbers represent delay at which promise to be resolved
-  // or rejected (-ve number)
-  // Third item in testcase row is the expected value or error message in string
-  // Last item is expected events
-  const testcases: [string, number[], (string | Error)[], RaceEvent[]][] = [
-    ["all resolve pre-cutoff", [100, 200], ["100", "200"], [resolved, resolved, precutoff]],
-    ["all resolve/reject pre-cutoff", [100, -200], ["100", "-200"], [resolved, rejected, precutoff]],
-    ["all reject pre-cutoff", [-100, -200], ["-100", "-200"], [rejected, rejected, precutoff]],
-    ["all reject pre-timeout", [-1100, -1200], ["-1100", "-1200"], [cutoff, rejected, rejected, pretimeout]],
-    ["race and resolve pre-timeout", [1100, 1200], ["1100", "pending"], [cutoff, resolved, pretimeout]],
```

Comment:
why have the raceevents been removed here?

Reply:
Mixing event callbacks with the Promise is not a good pattern, as you will review the new implementation is much clean and achieve what is needed without the events.